boards/tutorials/nrf52840dk-dynamic-apps-and-policies/src/main.rs

```diff
@@ -0,0 +1,649 @@
+    adc: &'static capsules_core::adc::AdcDedicated<'static, nrf52840::adc::Adc<'static>>,
+    led: &'static capsules_core::led::LedDriver<
+        'static,
+        kernel::hil::led::LedLow<'static, nrf52840::gpio::GPIOPin<'static>>,
+        4,
+    >,
+    alarm: &'static AlarmDriver,
+    scheduler: &'static RoundRobinSched<'static>,
+    screen: &'static ScreenDriver,
+    systick: cortexm4::systick::SysTick,
+    processes: &'static [Option<&'static dyn kernel::process::Process>],
+    process_info: &'static ProcessInfoDriver,
+    nonvolatile_storage: &'static IsolatedNonvolatileStorageDriver,
+    dynamic_app_loader: &'static AppLoaderDriver,
+}
+
+impl SyscallDriverLookup for Platform {
+    fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+    where
+        F: FnOnce(Option<&dyn kernel::syscall::SyscallDriver>) -> R,
+    {
+        match driver_num {
+            capsules_core::console::DRIVER_NUM => f(Some(self.console)),
+            capsules_core::alarm::DRIVER_NUM => f(Some(self.alarm)),
+            capsules_core::led::DRIVER_NUM => f(Some(self.led)),
+            capsules_core::button::DRIVER_NUM => f(Some(self.button)),
+            capsules_core::adc::DRIVER_NUM => f(Some(self.adc)),
+            capsules_extra::screen::DRIVER_NUM => f(Some(self.screen)),
+            capsules_extra::process_info_driver::DRIVER_NUM => f(Some(self.process_info)),
+            capsules_extra::isolated_nonvolatile_storage_driver::DRIVER_NUM => {
+                f(Some(self.nonvolatile_storage))
+            }
+            capsules_extra::app_loader::DRIVER_NUM => f(Some(self.dynamic_app_loader)),
+            _ => f(None),
+        }
+    }
+}
+
+/// This is in a separate, inline(never) function so that its stack frame is
+/// removed when this function returns. Otherwise, the stack space used for
+/// these static_inits is wasted.
+#[inline(never)]
+unsafe fn create_peripherals() -> &'static mut Nrf52840DefaultPeripherals<'static> {
+    let ieee802154_ack_buf = static_init!(
+        [u8; nrf52840::ieee802154_radio::ACK_BUF_SIZE],
+        [0; nrf52840::ieee802154_radio::ACK_BUF_SIZE]
+    );
+    // Initialize chip peripheral drivers
+    let nrf52840_peripherals = static_init!(
+        Nrf52840DefaultPeripherals,
+        Nrf52840DefaultPeripherals::new(ieee802154_ack_buf)
+    );
+
+    nrf52840_peripherals
+}
+
+impl KernelResources<nrf52840::chip::NRF52<'static, Nrf52840DefaultPeripherals<'static>>>
+    for Platform
+{
+    type SyscallDriverLookup = Self;
+    type SyscallFilter = ();
+    type ProcessFault = ();
+    type Scheduler = RoundRobinSched<'static>;
+    type SchedulerTimer = cortexm4::systick::SysTick;
+    type WatchDog = ();
+    type ContextSwitchCallback = ();
+
+    fn syscall_driver_lookup(&self) -> &Self::SyscallDriverLookup {
+        self
+    }
+    fn syscall_filter(&self) -> &Self::SyscallFilter {
+        &()
```

Comment:
```suggestion
        self.base.syscall_filter()
```

While the effect is the same at the moment (since the base board defines this as `&()`), I think you want to [follow the pattern of other 'sub-boards'](https://github.com/tock/tock/blob/a6237dac5021a0b341699172525892f09237a68f/boards/tutorials/nrf52840dk-thread-tutorial/src/main.rs#L75-L96) and derive this [and the rest here] from the base, for this and the rest of the platform methods.